parquet-common/src/main/java/org/apache/parquet/bytes/CapacityByteArrayOutputStream.java

```diff
@@ -220,6 +221,11 @@ public void write(byte b[], int off, int len) {
       currentSlabIndex += len;
       bytesUsed += len;
     }
+    if (bytesUsed < 0) {
```

Comment:
Is it better to proactively check the overflow before addition?

Reply:
Thanks, I moved the check for overflow to before starting to write.